tests/401_raw_identifier.rs

```diff
@@ -0,0 +1,83 @@
+use ron::error::{Error, Position, SpannedError};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename = "Hello World")]
+struct InvalidStruct;
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename = "")]
+struct EmptyStruct;
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+#[serde(rename = "Hello+World")]
+struct RawStruct {
+    #[serde(rename = "ab.cd-ef")]
+    field: bool,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+enum RawEnum {
+    #[serde(rename = "Hello-World")]
+    RawVariant,
+}
+
+#[test]
+fn test_invalid_identifiers() {
+    let ser = ron::ser::to_string_pretty(
+        &InvalidStruct,
+        ron::ser::PrettyConfig::default().struct_names(true),
+    );
+    assert_eq!(
+        ser,
+        Err(Error::InvalidIdentifier(String::from("Hello World")))
+    );
+
+    let ser = ron::ser::to_string_pretty(
+        &EmptyStruct,
+        ron::ser::PrettyConfig::default().struct_names(true),
+    );
+    assert_eq!(ser, Err(Error::InvalidIdentifier(String::from(""))));
+
+    let de = ron::from_str::<InvalidStruct>("Hello World").unwrap_err();
+    assert_eq!(
+        de,
+        SpannedError {
+            code: Error::ExpectedDifferentStructName {
+                expected: "Hello World",
+                found: String::from("Hello"),
+            },
+            position: Position { line: 1, col: 6 },
+        }
+    );
+
+    let de = ron::from_str::<EmptyStruct>("").unwrap_err();
```

Comment:
Can we add a test for an empty raw identifier? Would that get accepted?

Reply:
Done ... though this exploded a bit in d85ca35a13bf2ebccd16237479adfafc8293182f and 37df3d8f6a17b3adf0ee0b2fbf668d972710c80b

Dealing with empty (raw) identifiers is tricky. The solution now is the following:
- if the expected identifier is valid, we complain about not finding the opening `(` for the expected named struct
- if the expected identifier is empty, the same error struct is used but the message is adapeted to the following: `1:1: Expected only opening `(`, no name, for un-nameable struct`
- if the expected identifier is invalid in another way, the usual error message (as in case 1) is given, but the expected struct name is printed as `"ö"_[invalid identifier]`

I've added this new identifier formatting to error messages in general, such that struct, variant, and field names are now always automatically printed either as \`id`, \`r#i+d`, or "ö"_[invalid identifier]

Reply:
This now also does a tiny bit of cleanup to the `deserialize_struct` method by pulling out everything after the struct name check, so that we no longer pass `""` as an identifier to the method internally, meaning it can now complain about it as an error instead of having to assume it was our internal code that passed it in